src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTestData.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace System.Security.Cryptography.SLHDsa.Tests
+{
+    public static class SlhDsaTestData
+    {
+        /// <summary>
+        /// Gets the negation of <see cref="SlhDsa.IsSupported"/>. This can be used to conditionally skip tests.
+        /// </summary>
+        public static bool IsNotSupported => !SlhDsa.IsSupported;
+
+        public static IEnumerable<object[]> AlgorithmsData => AlgorithmsRaw.Select(a => new[] { a });
+
+        public static SlhDsaAlgorithm[] AlgorithmsRaw =
+        [
+            SlhDsaAlgorithm.SlhDsaSha2_128s,
+            SlhDsaAlgorithm.SlhDsaShake128s,
+            SlhDsaAlgorithm.SlhDsaSha2_128f,
+            SlhDsaAlgorithm.SlhDsaShake128f,
+            SlhDsaAlgorithm.SlhDsaSha2_192s,
+            SlhDsaAlgorithm.SlhDsaShake192s,
+            SlhDsaAlgorithm.SlhDsaSha2_192f,
+            SlhDsaAlgorithm.SlhDsaShake192f,
+            SlhDsaAlgorithm.SlhDsaSha2_256s,
+            SlhDsaAlgorithm.SlhDsaShake256s,
+            SlhDsaAlgorithm.SlhDsaSha2_256f,
+            SlhDsaAlgorithm.SlhDsaShake256f,
+        ];
+
+        // TODO: Construction tests for OpenSsl, and add hierarchy
+        // SlhDsaTemplateTests -> SlhDsaTestImplTests (use test implementation)
+        //                     -> SlhDsaImplTests -> SlhDsaOpenSslTests (use safe handle to construct)
+        //                                        -> SlhDsaDefaultImplTests (use static factory Api to construct)
+
+        public record SlhDsaTestVector(SlhDsaAlgorithm Algorithm, string SecretKey, string PublicKey, string Message, string Context, string Signature);
+
+        // The vectors below are generated from the NIST ACVP projections.
+        // https://raw.githubusercontent.com/usnistgov/ACVP-Server/refs/heads/master/gen-val/json-files/SLH-DSA-sigGen-FIPS205/internalProjection.json
+        public static IEnumerable<SlhDsaTestVector> NistSignTestVectors =
```

Comment:
We will, at some point, want at least one known test for every SLH-DSA parameter set, just to make sure we do plumb everything through correctly.